pkg/cmd/get/probe.go

```diff
@@ -0,0 +1,303 @@
+			utils.White_B.Print("\nEnter the Project ID: ")
+			fmt.Scanln(&projectID)
+
+			if projectID == "" {
+				utils.Red.Println("⛔ Project ID can't be empty!!")
+				os.Exit(1)
+			}
+		}
+		ProbeID, err := cmd.Flags().GetString("probe-id")
+
+		if ProbeID == "" {
+			getProbeList(projectID, cmd, credentials)
+		} else {
+			getProbeDetails(projectID, ProbeID, credentials)
+
+		}
+
+	},
+}
+
+func init() {
+	GetCmd.AddCommand(probesCmd)
+
+	probesCmd.Flags().String("project-id", "", "Set the project-id to get Probe from a particular project.")
+	probesCmd.Flags().BoolP("non-interactive", "n", false, "Set it to true for non interactive mode | Note: Always set the boolean flag as --non-interactive=Boolean")
+	probesCmd.Flags().String("probe-types", "", "Set the probe-types as comma separated values to filter the probes")
+	probesCmd.Flags().String("probe-id", "", "Set the probe-details to the ID of probe for getting all the details related to the probe.")
+}
+
+func getProbeList(projectID string, cmd *cobra.Command, credentials types.Credentials) {
+	var selectedItems []*models.ProbeType
+	NoninteractiveMode, err := cmd.Flags().GetBool("non-interactive")
+	utils.PrintError(err)
+
+	if NoninteractiveMode == false {
+		prompt := promptui.Select{
+			Label: "Do you want to enable advance filter probes?",
+			Items: []string{"Yes", "No"},
+		}
+		_, option, err := prompt.Run()
+		if err != nil {
+			fmt.Printf("Prompt failed %v\n", err)
+			return
+		}
+		fmt.Printf("You chose %q\n", option)
+
+		if option == "Yes" {
+			items := []models.ProbeType{"httpProbe", "cmdProbe", "promProbe", "k8sProbe", "done"}
+			for {
+				prompt := promptui.Select{
+					Label: "Select ProbeType",
+					Items: items,
+					Templates: &promptui.SelectTemplates{
+						Active:   `▸ {{ . | cyan }}`,
+						Inactive: `  {{ . | white }}`,
+						Selected: `{{ "✔" | green }} {{ . | bold }}`,
+					},
+				}
+
+				selectedIndex, result, err := prompt.Run()
+				if err != nil {
+					fmt.Printf("Prompt failed %v\n", err)
+					os.Exit(1)
+				}
+
+				if items[selectedIndex] == "done" {
+					break
+				}
+
+				final := models.ProbeType(result)
+				selectedItems = append(selectedItems, &final)
+				items = append(items[:selectedIndex], items[selectedIndex+1:]...)
+
+			}
+
+			fmt.Printf("Selected Probe Types: %v\n", selectedItems)
+		}
+	} else {
+		var probeTypes string
+		probeTypes, err = cmd.Flags().GetString("probe-types")
+		values := strings.Split(probeTypes, ",")
+		for _, value := range values {
+			probeType := models.ProbeType(value)
+			selectedItems = append(selectedItems, &probeType)
+		}
+	}
+
+	probes_get, _ := apis.ListProbeRequest(projectID, selectedItems, credentials)
+	probes_data := probes_get.Data.Probes
+
+	itemsPerPage := 5
+	page := 1
+	totalProbes := len(probes_data)
+
+	writer := tabwriter.NewWriter(os.Stdout, 8, 8, 8, '\t', tabwriter.AlignRight)
+	utils.White_B.Fprintln(writer, "PROBE ID\t PROBE TYPE\t CREATED BY\t CREATED AT")
```

Comment:
Let's have `referencedBy` also here ( as we also show this UI Overview )

Reply:
done you can take a look